### ISSUE TITLE
Pop out window for the web editor

### ISSUE THREAD

**Issue by aahdee** [processing/p5.js-web-editor#1651]
#### Nature of issue?

- New feature request

#### New feature details:

Sometimes one would like to create sketches that are large in size. But if one does so, in order to see the entire sketch one must squish their code to the left and essentially make it unreadable. Could there be an option to open the sketch in a pop up window so that you can view your code and sketch at the same time?

**Comment by catarak** [comment]
Thanks for the suggestion! I think this is a great idea. The infrastructure that would take to build this is super related to #1337, which is about adding a collaborative live coding mode. I think to make this happen the sketches would need to update with CRDT's and we'd need to add web sockets.

**Comment by vulongphan** [comment]
Hi @catarak, I would like to work on this issue. I really think that this is an important feature to be implemented since the same feature is very useful in desktop app Processing.  

At the first glance, I believe an intuitive approach would be to figure out how to create a pop-up window and then how to sync the sketch in the preview section with the sketch in the pop-up window. How do you think about this approach and could you also provide some materials that would assist me in building this feature?

Thank you!

**Comment by catarak** [comment]
I agree that this would be a great feature to have! Before it can happen, I think some infrastructure needs to happen first, specifically, moving sketch updates to use CRDT's (see some of the discussion in #1337).

**Comment by vulongphan** [comment]
I figure in order to change how the sketches are saved and to apply CRDT I need to spend more time researching and to have a better understanding of the codebase. In the meantime, anyone can freely work on this issue!

**Comment by catarak** [comment]
I'm working on #166 which also is a stepping stone to making this issue happen!

**Comment by jgrizou** [comment]
I developed a small webapp, called [p5live](https://p5live.jgrizou.repl.co/), that should help evaluate what having a pop up window feels like, see #1817

Or read more at https://twitter.com/jgrizou/status/1371882296375132162
